<launch>
    <!-- <arg name="image_topic" default="" /> -->

    <!-- <arg name="camera_id" default="3" />
    <arg name="target_tag_ids" default="[0, 1, 2, 3]" /> -->
    <!-- <arg name="target_tag_ids" default="[0]" /> -->

    <!-- <arg name="camera_id" default="2" />
    <arg name="target_tag_ids" default="[4, 5, 6, 7]" /> -->

    <arg name="camera_id" default="1" />
    <arg name="target_tag_ids" default="[2, 3, 4, 5]" />

    <!-- <arg name="camera_id" default="0" />
    <arg name="target_tag_ids" default="[6, 7, 8, 9]" /> -->

    <node name="image_undistortion_node" pkg="dots_ucs" type="image_undistortion_node.py" output="screen">
        <param name="camera_id" value="$(arg camera_id)" />
        <param name="image_topic" value="/camera$(arg camera_id)/image_raw" />
        <param name="camera_info_file" value="$(find dots_ucs)/config/camera$(arg camera_id)_params.yaml" />
        <param name="apriltag_info_file" value="$(find dots_ucs)/config/apriltags.yaml" />
        <param name="output_topic" value="/camera$(arg camera_id)/image_apriltag" />
        <param name="undistorted_topic" value="/camera$(arg camera_id)/image_undistorted" />
        <param name="top_view_topic" value="/camera$(arg camera_id)/image_top_view" />
        <param name="target_tag_ids" value="$(arg target_tag_ids)" />
    </node>
    
    <!-- <param name="image_topic" value="/camera$(arg camera_id)/image_undistorted" /> -->
    <!-- <param name="image_topic" value="/camera$(arg camera_id)/image_raw" /> -->
    <node pkg="dots_ucs" type="circle_grid_auto_pattern.py" name="circle_grid_auto_pattern" output="screen">
        <param name="image_topic" value="/camera$(arg camera_id)/image_top_view" />
    </node>
    
</launch>
